src/Symfony/Component/AssetMapper/ImportMap/ImportMapManager.php

```diff
@@ -0,0 +1,399 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\AssetMapper\ImportMap;
+
+use Symfony\Component\AssetMapper\AssetDependency;
+use Symfony\Component\AssetMapper\AssetMapper;
+use Symfony\Component\HttpClient\HttpClient;
+use Symfony\Component\VarExporter\VarExporter;
+use Symfony\Contracts\HttpClient\HttpClientInterface;
+
+/**
+ * @experimental
+ *
+ * @author Kévin Dunglas <kevin@dunglas.dev>
+ * @author Ryan Weaver <ryan@symfonycasts.com>
+ * @final
+ */
+class ImportMapManager
+{
+    public const PROVIDER_JSPM = 'jspm';
+    public const PROVIDER_JSPM_SYSTEM = 'jspm.system';
+    public const PROVIDER_SKYPACK = 'skypack';
+    public const PROVIDER_JSDELIVR = 'jsdelivr';
+    public const PROVIDER_UNPKG = 'unpkg';
+    public const PROVIDERS = [
+        self::PROVIDER_JSPM,
+        self::PROVIDER_JSPM_SYSTEM,
+        self::PROVIDER_SKYPACK,
+        self::PROVIDER_JSDELIVR,
+        self::PROVIDER_UNPKG,
+    ];
+
+    public const POLYFILL_URL = 'https://ga.jspm.io/npm:es-module-shims@1.7.2/dist/es-module-shims.js';
+
+    /**
+     * @see https://regex101.com/r/2cR9Rh/1
+     *
+     * Partially based on https://github.com/dword-design/package-name-regex
+     */
+    private const PACKAGE_PATTERN = '/^(?:https?:\/\/[\w\.-]+\/)?(?:(?<registry>\w+):)?(?<package>(?:@[a-z0-9-~][a-z0-9-._~]*\/)?[a-z0-9-~][a-z0-9-._~]*)(?:@(?<version>[\w\._-]+))?(?:(?<subpath>\/.*))?$/';
+    public const IMPORT_MAP_FILE_NAME = 'importmap.json';
+
+    private array $importMapEntries;
+    private array $modulesToPreload;
+    private string $json;
+
+    public function __construct(
+        private readonly AssetMapper $assetMapper,
+        private readonly string      $importMapConfigPath,
+        private readonly string      $vendorDir,
+        private readonly string      $provider = self::PROVIDER_JSPM,
```

Comment:
we don't align `$`, that'd require a php-cs-fixer rule if we start doing so
for now, we should keep consistency I suppose

Reply:
We explicit rejected aligning `=>` in multiline arrays years ago, to avoid diffs re-indenting the whole array (and causing git conflicts) any time a longer key is added. I'm very strongly in favor of not aligning the `$` either for promoted properties for the same reason.

Reply:
np - something in my toolset did this for me :)